packages/react/toggle-group/src/ToggleGroup.tsx

```diff
@@ -227,48 +250,41 @@ type ToggleGroupItemPrimitive = Polymorphic.ForwardRefComponent<
 >;
 
 const ToggleGroupItem = React.forwardRef((props, forwardedRef) => {
+  const valueContext = useToggleGroupValueContext(ITEM_NAME);
   const context = useToggleGroupContext(ITEM_NAME);
-  return context.rovingFocus ? (
-    <ToggleGroupRovingFocusItem {...props} ref={forwardedRef} />
-  ) : (
-    <ToggleGroupItemImpl {...props} ref={forwardedRef} />
-  );
+  const pressed = valueContext.value.includes(props.value);
+  const disabled = context.disabled || props.disabled;
+  const Comp = context.rovingFocus ? ToggleGroupRovingFocusItem : ToggleGroupItemImpl;
```

Comment:
We've avoided naming these sorts of conditionals `Comp` in the past I believe because it clashes with when we do `as: Comp = DEFAULT_TAG`. 

For a second I thought that this wasn't using `Primitive` but it is.

Reply:
Would you rather I keep the conditional inline and duplicate the ref/pressed/disabled props?

Reply:
Yeah I'm okay with that duplication in this instance because it's super simple, nothing complex going on there.